vnext/Shared/WinRTHttpResource.h

```diff
@@ -0,0 +1,73 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+#pragma once
+
+#include <IHttpResource.h>
+
+// Windows API
+#include <winrt/Windows.Web.Http.h>
+
+// Standard Library
+#include <mutex>
+
+namespace Microsoft::React {
+
+class WinRTHttpResource : public IHttpResource, public std::enable_shared_from_this<WinRTHttpResource> {
+  typedef winrt::Windows::Foundation::IAsyncOperationWithProgress<
+      winrt::Windows::Web::Http::HttpResponseMessage,
+      winrt::Windows::Web::Http::HttpProgress>
+      ResponseType;
+
+  static int64_t s_lastRequestId;
+
+  winrt::Windows::Web::Http::IHttpClient m_client;
+  std::mutex m_mutex;
+  std::unordered_map<int64_t, ResponseType> m_requests;
```

Comment:
Note, will rename. These are **not** request per-se, but rather response "promises" (WinRT async operations).